internal/parser/parser.go

```diff
@@ -89,23 +97,49 @@ func (p *Parser) parseLinks(s string) []string {
 	var result []string
 	links := p.linkRegex.FindAllString(s, -1)
 	for _, link := range links {
-		start := strings.IndexRune(link, '(') + len(string("("))
+		start := strings.IndexRune(link, '(') + len("(")
 		end := strings.IndexRune(link, ')')
 		result = append(result, link[start:end])
 	}
 	return result
 }
 
 func parseSection(section, s string) string {
+	const (
+		sectionEnd = "#"
+		blockDelim = "```"
+	)
+
 	start := strings.Index(s, section)
 	if start == -1 {
 		return ""
 	}
-	const end = "#"
+
 	s = s[start+len(section):]
-	e := strings.Index(s, end)
-	if e == -1 {
-		return s
+
+	end := -1
+	for blockEnd := 0; blockEnd > end; {
+		if end = strings.Index(s[blockEnd:], sectionEnd); end == -1 {
+			return s
+		}
+		end += blockEnd
+
+		for {
+			blockStart := strings.Index(s[blockEnd:], blockDelim)
+			if blockStart < 0 {
+				break
+			}
```

Comment:
This can be a part of for loop,
Also, consider moving this for loop to private func.

Reply:
Yep, extracted inner loop to `skipBlocks()`.